unified_planning/plans/partial_order_plan.py

```diff
@@ -0,0 +1,70 @@
+# Copyright 2021 AIPlan4EU project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+import networkx as nx # type: ignore
+import unified_planning as up
+import unified_planning.plans as plans
+from unified_planning.plans.sequential_plan import SequentialPlan
+from typing import Callable, Dict, Iterator, List
+
+
+class PartialOrderPlan(plans.plan.Plan):
+    '''Represents a partial order plan. Actrions are represent as an adjagency list graph.'''
+    def __init__(self, actions: Dict['plans.plan.ActionInstance', List['plans.plan.ActionInstance']]):
+        self._graph = nx.convert.from_dict_of_lists(actions, create_using=nx.DiGraph)
+
+    def __repr__(self) -> str:
+        return str(self._graph )
+
+    def __eq__(self, oth: object) -> bool:
+        if isinstance(oth, PartialOrderPlan):
+            adj_dict = self.actions
```

Comment:
Modulo the comment below this looks OK. But a much simple alternative would be to use `is_isomorphic` [1] (if you specify a `node_match`, in our case the complexity should be O(V^2))

[1] https://networkx.org/documentation/stable/reference/algorithms/generated/networkx.algorithms.isomorphism.is_isomorphic.html